### PR TITLE
Add method to save snapshots

### LINKED ISSUE #4: Saving images to disk
I'm currently looking into different pixelflut implementations, and I like yours a lot. I am missing one thing so far, that lets me tend to the python implementation. I was wondering if this could be implemented.

This python implementation can periodically (i think every second) store the image to disk. This is very useful to me, because I create timelapse-videos after a pixelflut session.

https://github.com/defnull/pixelflut/blob/3458157a242ba1789de7ce308480f4e1cbacc916/pixelflut/pixelflut.py#L188-L190

If there's a possibility to have this in rust as well, I'd be delighted!

### PR DESCRIPTION
**Pull request by datdenkikniet**
Fix #4 

Depends on https://github.com/timvisee/pixelpwnr-render/pull/4

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -95,6 +106,35 @@ async fn listen(listener: std::net::TcpListener, pixmap: Arc<Pixmap>, stats: Arc
...
+
+        image::save_buffer(
+            path,
+            image_data,
+            width as u32,
+            height as u32,
+            image::ColorType::RGBA(8),
+        )
+        .unwrap();
+        tokio::time::sleep(interval).await;
```
Comment (timvisee) [inline src/main.rs:134]: Though this is fine, you can also use [`tokio::time::interval`](https://docs.rs/tokio/latest/tokio/time/fn.interval.html).

**Review by timvisee — APPROVED** [review]

**Comment by timvisee** [comment]
Fantastic work once again.

Please see [my comment](https://github.com/timvisee/pixelpwnr-server/pull/6#discussion_r949100172). I'll merge this now, but it might be worth to change that.

Note that I might merge `pixelpwnr-render` into this repository, as it doesn't make much sense to keep those separated. What do you think?

File: src/main.rs (hunk 1 above)
Comment (datdenkikniet) [reply src/main.rs:134]: Aha, never saw that one before! That's a lot neater. It probably doesn't make a terribly great difference here, but good to know and we should definitely change that in the future.

**Comment by datdenkikniet** [comment]
Combining them into the same repo makes sense, yeah. The two crates are interlinked pretty heavily, so having them in the same repository, within one workspace, would make sense. The only downside is that doing releases is a little trickier, but since the crate isn't on `crates.io` (yet?) anyways that doesn't make a difference here.

**Comment by timvisee** [comment]
No, it isn't on `crates.io` yet. I'll release it on there some time.

Releasing a single-repo-multi-crate setup isn't hard. I'm also doing that in [`prs`](https://github.com/timvisee/prs) for example.

**Comment by timvisee** [comment]
Trying to merge `pixelpwnr-render` into this right now on [`merge-pixelpwnr-render`](https://github.com/timvisee/pixelpwnr-server/tree/merge-pixelpwnr-render).

**Comment by timvisee** [comment]
`pixelpwnr-render` has now been merged into this repository on the master branch.

**Comment by datdenkikniet** [comment]
Nice! I was working on some additional stuff, let's see how I can trick `git-subtree` into doing it for me without losing too much history :P

**Comment by timvisee** [comment]
> Nice! I was working on some additional stuff, let's see how I can trick `git-subtree` into doing it for me without losing too much history :P

Sorry, I wasn't aware of that. The two major changes are that `-server` and `-render` sources moved into `server` and `render` subdirectories.

Generating a patch for each changed file, and applying it again on the new location (after pulling) might do the trick.

**Comment by datdenkikniet** [comment]
No worries, you didn't do anything unexpected :P The changes were too big to merge before this properly anyways, and I don't mind, actually thought it was a little interesting to see if I could get it to work. Progress on the [`compression`](https://github.com/datdenkikniet/pixelpwnr-server/tree/compression) branch! 

There are also some features on there that probably deserve their own PR, namely not-entirely-atomic pixelcount updates and an "n pixels" binary command but :shrug:

**Pull request merged**